backend/typescript/rest/schedulingRoutes.ts

```diff
@@ -0,0 +1,144 @@
+import { Router } from "express";
+
+import { isAuthorizedByRole } from "../middlewares/auth";
+import {
+  createSchedulingDtoValidator,
+  updateSchedulingDtoValidator,
+} from "../middlewares/validators/schedulingValidators";
+import nodemailerConfig from "../nodemailer.config";
+import AuthService from "../services/implementations/authService";
+import EmailService from "../services/implementations/emailService";
+import UserService from "../services/implementations/userService";
+import SchedulingService from "../services/implementations/schedulingService";
+import IUserService from "../services/interfaces/userService";
+import IAuthService from "../services/interfaces/authService";
+import IEmailService from "../services/interfaces/emailService";
+import ISchedulingService from "../services/interfaces/schedulingService";
+import { SchedulingDTO } from "../types";
+import { sendResponseByMimeType } from "../utilities/responseUtil";
+
+const schedulingRouter: Router = Router();
+
+// Commenting out authorization for now
+// schedulingRouter.use(isAuthorizedByRole(new Set(["Admin"])));
+
+const userService: IUserService = new UserService();
+const emailService: IEmailService = new EmailService(nodemailerConfig);
+const authService: IAuthService = new AuthService(userService, emailService);
+const schedulingService: ISchedulingService = new SchedulingService();
+
+/* Get all schedulings, optionally filter by:
+  - id, through URI (ex. /scheduling/1)
+  - donorId, through query param (ex. /scheduling/?donorId=1)
+*/
+schedulingRouter.get("/:id?", async (req, res) => {
+  const id: number | null = Number(req.params.id);
```

Comment:
what happens when id or donorId are not passed in as a number. For example in cases like `/scheduling/hello` or `/scheduling/?donorId=1=hello`

Reply:
Actually can we do it how its done in the donor PR? `id` and `donorId` stays as a string when passed in the schedulingService functions, and then converted to Number when calling the database. It would make more sense if you take a quick look at donor PR on how this is implemented.

Reply:
updated!